main.go

```diff
@@ -209,32 +301,54 @@ func run() error {
 			continue
 		}
 
-		rows = append(rows, generateRow(baseRef, prevBench))
-
-		var ratioNsPerOp float64
-		if prevBench.NsPerOp != 0 {
-			ratioNsPerOp = (headBench.NsPerOp - prevBench.NsPerOp) / prevBench.NsPerOp
+		getRationsPerOP := func(headBench, baseBench *parse.Benchmark) (ratioNsPerOp float64) {
+			if prevBench.NsPerOp != 0 {
+				ratioNsPerOp = (headBench.NsPerOp - baseBench.NsPerOp) / baseBench.NsPerOp
+			}
+			return
 		}
 
-		var ratioAllocedBytesPerOp float64
-		if prevBench.AllocedBytesPerOp != 0 {
-			ratioAllocedBytesPerOp = (float64(headBench.AllocedBytesPerOp) - float64(prevBench.AllocedBytesPerOp)) / float64(prevBench.AllocedBytesPerOp)
+		getRatioAllocedBytesPerOp := func(headBench, baseBench *parse.Benchmark) (ratioAllocedBytesPerOp float64) {
+			if prevBench.AllocedBytesPerOp != 0 {
+				ratioAllocedBytesPerOp = (float64(headBench.AllocedBytesPerOp) - float64(baseBench.AllocedBytesPerOp)) / float64(baseBench.AllocedBytesPerOp)
+			}
+			return
 		}
 
+		rows = append(rows, generateRow(baseRef, prevBench))
 		ratios = append(ratios, result{
 			Benchmark:              benchmark,
-			RatioNsPerOp:           ratioNsPerOp,
-			RatioAllocedBytesPerOp: ratioAllocedBytesPerOp,
+			RatioNsPerOp:           getRationsPerOP(headBench, prevBench),
+			RatioAllocedBytesPerOp: getRatioAllocedBytesPerOp(headBench, prevBench),
 		})
+
+		// get benchmark result of latestReleaseVersion
+		if latestReleaseSet == nil {
+			continue
+		}
+		if latestReleaseBench, ok := latestReleaseSet[benchName]; ok {
+			rows = append(rows, generateRow(tagName, latestReleaseBench))
+			ratiosWithRelease = append(ratiosWithRelease, result{
+				Benchmark:              benchmark,
+				RatioNsPerOp:           getRationsPerOP(headBench, latestReleaseBench),
+				RatioAllocedBytesPerOp: getRatioAllocedBytesPerOp(headBench, latestReleaseBench),
+			})
+		}
 	}
 
 	if !onlyRegression {
 		showResult(os.Stdout, rows)
 	}
 
-	regression := showRatio(os.Stdout, ratios, onlyRegression)
-	if regression {
-		return fmt.Errorf("This commit makes benchmarks worse")
+	regression := showRatio(os.Stdout, ratios, onlyRegression, baseRef)
+
+	var regressionWithLatestVersion bool
+	if latestReleaseSet != nil {
+		regressionWithLatestVersion = showRatio(os.Stdout, ratiosWithRelease, onlyRegression, tagName)
+	}
+	if regression || regressionWithLatestVersion {
+		return fmt.Errorf("this commit makes benchmarks worse，compared with %s: %t, comapred with %s: %t",
+			baseRef, regression, tagName, regressionWithLatestVersion)
```

Comment:
this is not clear (there is also a typo at "comapred")

you should either separate the 2 checks (`regression` / `regressionWithLatestVersion`), or change the error message

Reply:
Separate the 2 checks seems to make the code a little tediously long.
The 'true' or 'false' means which commit makes the benchmark result worse.
```
F1031 18:23:03.160775    1692 main.go:67] this commit makes benchmarks worse，compared with HEAD~1: false, compared with refs/tags/v1.3.0: true
```